### PR TITLE
Make duckdb a singleton

### PR DESCRIPTION
**Pull request by JelteF**
Based on work by @Y-- and PR #102 and #147. This basically extracts the work from there into a separate PR, so it can be merged before some of the open TODOs related to prepared statements/extended protocol are resolved.

### CODE REVIEW

**Review by mkaruza — APPROVED** [review]

File: include/pgduckdb/scan/postgres_scan.hpp [hunk 1]
```diff
@@ -39,14 +39,14 @@ class PostgresScanLocalState {
 	bool m_exhausted_scan;
 };
 
-struct PostgresReplacementScanData : public duckdb::ReplacementScanData {
+struct PostgresReplacementScanDataClientContextState : public duckdb::ClientContextState {
```
Comment (Tishj) [inline include/pgduckdb/scan/postgres_scan.hpp:42]: Can we shorten this to `PostgresReplacementState`?

Also for future reference, I think a good portion of these class variables are unused, such as `m_rtables`

File: include/pgduckdb/pgduckdb_duckdb.hpp [hunk 2]
```diff
@@ -10,7 +10,30 @@ extern "C" {
...
+public:
+	static inline const DuckDBManager &
+	Get() {
+		static DuckDBManager instance;
+		return instance;
+	}
+
+	inline duckdb::DuckDB &
+	GetDatabase() const {
+		return *database;
```
Comment (Tishj) [inline include/pgduckdb/pgduckdb_duckdb.hpp:23]: Can we do a nullcheck here? I imagine this just causes a generic looking InternalException if `database` is null which will be hard to debug
